File: frontend/src/types/auditLog.ts
export interface AuditLog {
  id: number;
  user_id: number | null;
  action: string;
  entity_type: string | null;
  entity_id: number | null;
  description: string | null;
  created_at: string;
}

export interface AuditLogListResponse {
  logs: AuditLog[];
  total: number;
  page: number;
  page_size: number;
  total_pages: number;
}

export interface AuditLogQueryParams {
  page?: number;
  page_size?: number;
  action?: string;
  entity_type?: string;
  user_id?: number;
}

File: backend/app/schemas/audit_log.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class AuditLogResponse(BaseModel):

    id: int

    user_id: int | None

    action: str

    entity_type: str | None

    entity_id: int | None

    description: str | None

    created_at: datetime

    model_config = ConfigDict(
        from_attributes=True
    )


class AuditLogListResponse(BaseModel):

    logs: list[AuditLogResponse]

    total: int

    page: int

    page_size: int

    total_pages: int